src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,107 @@
+package duke;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Gets tasks from file, creates a new file if file is absent, and upload tasks in file.
+ */
+public class Storage {
+    private String pathname;
+    private ArrayList<Task> listStored;
+    private String directory;
+
+    /**
+     * Creates a Storage object.
+     * @param pathname Pathname to the file to be stored.
+     * @param listStored
+     * @param directory Directory of the file to be stored.
+     */
+    Storage(String pathname, ArrayList<Task> listStored, String directory) {
+        this.pathname = pathname;
+        this.listStored = listStored;
+        this.directory = directory;
+    }
+
+    /**
+     * Gets data from the given path of the file and store the tasks in this.listStored.
+     */
+    public void getData() {
+        try {
+            File f = new File(this.pathname);
+            Scanner s = new Scanner(f);
+            while (s.hasNext()) {
+                String string = s.nextLine();
+                String typeOfTask = string.substring(1, 2);
+                boolean gotMarkSymbol = string.substring(4, 5).equals("X");
+
+                if (typeOfTask.equals("T")) {
+                    if (gotMarkSymbol) {
+                        listStored.add(new ToDo(string.substring(7), true));
+                    } else {
+                        listStored.add(new ToDo(string.substring(6), false));
+                    }
+                } else {
+                    int openBracketIndex = string.indexOf("(");
+                    int closeBracketIndex = string.indexOf(")");
+                    String nameOfMarkedTaskForDeadlineOrEventTask = string.substring(7, openBracketIndex - 1);
```

Comment:
Maybe ```nameOfMarkedTaskForDeadlineOrEventTask``` can be renamed to something shorter like ```markedDeadlineOrEventName``` so that it is easier to read?